client/jetpack-onboarding/steps/summary.jsx

```diff
@@ -41,11 +48,21 @@ class JetpackOnboardingSummaryStep extends React.PureComponent {
 				<div className="steps__summary-columns">
 					<div className="steps__summary-column">
 						<h3 className="steps__summary-heading">{ translate( "Steps you've completed:" ) }</h3>
-						<CompletedSteps siteId={ siteId } siteSlug={ siteSlug } steps={ steps } />
+						<CompletedSteps
+							handleNextStepSelection={ this.handleNextStepSelection }
```

Comment:
This would be called when there's a click on a completed step - I feel it's a little odd to name it like this. 

This also leads to the next one: perhaps it would make sense to be able to differentiate between whether a completed step link or a next step link has been clicked (by having two separate events)?

Naturally, if we have separate events for those separate cases, maybe it would make sense to declare their handlers directly in the responsible components? This way we won't need to pass them down as props.